runtime/marshal_jsonpb.go

```diff
@@ -151,7 +151,15 @@ func (d DecoderWrapper) Decode(v interface{}) error {
 
 // NewEncoder returns an Encoder which writes JSON stream into "w".
 func (j *JSONPb) NewEncoder(w io.Writer) Encoder {
-	return EncoderFunc(func(v interface{}) error { return j.marshalTo(w, v) })
+	return EncoderFunc(func(v interface{}) error {
+		if err := j.marshalTo(w, v); err != nil {
+			return err
+		}
+		// mimic json.Encoder by adding a newline (makes output
+		// easier to read when it contains multiple encoded items)
+		_, err := w.Write([]byte("\n"))
+		return err
+	})
```

Comment:
Could we use `j.Delimiter` instead?

Reply:
Let me look. I think when I originally prepared this patch (in a fork of the repo that we're using at FullStory), it may not have had `Delimiter`. (So maybe this PR is no longer needed; I'll dig around and either update it or close it.)

Reply:
So, from looking a little more closely, an optional `Delimiter()` method can be added to control the output of server-streaming methods.

However, the `JSONPb` type itself has no field named `Delimiter`. And, IMO, it does not need to be configurable but rather should just mirror `JSONBuiltin` (whose `Encoder` already adds newlines in between messages; and it also implements `Delimiter()` to return a newline).

So I think this part of the patch is still applicable. But that I also should add a `Delimiter()` method to this type, to mirror `JSONBuiltin`.